shortcuts.go

```diff
@@ -79,14 +80,14 @@ func GetShortcuts() ([]Shortcut, error) {
 	return shortcuts, nil
 }
 
-func CheckNameDuplication(name *string) error {
+func CheckNameDuplication(category *string, name *string) error {
 
 	shortcutList, _ := GetShortcuts()
 	for _, shortcut := range shortcutList {
```

Comment:
Performance nit: Looks like this should be a map (`O(1)`) (or two?) instead of a slice (`O(n)`).
This will probably have to be reflected in the JSON as well.
I.e. instead of:
```json
{
	"shortcuts": [{"name": "...", "category": "..."}]
}
```
we do
```json
{
	"shortcuts": {
		"shell": {
			"cancel": "ctrl+c",
			"background": "ctrl+z",
			"etc.": "etc."
		}
	}
}
```

Reply:
Thank you!
Yes, now this should be with map to make it simpler.